From 508c0d47736023fdcb491fd55ad64f7e08f6f32f Mon Sep 17 00:00:00 2001
From: Paul Walmsley <paul@pwsan.com>
Date: Mon, 29 Oct 2012 20:49:43 -0600
Subject: [PATCH 310/951] ARM: OMAP1: CGRM: fix omap1_get_reset_sources()
 return type

An older version of the patch "ARM: OMAP1: create read_reset_sources()
function (for initial use by watchdog)" was sent upstream, which used
the wrong return type for the omap1_get_reset_sources() function.
Fix it to return a u32, which is what the WDTIMER platform_data
function pointer read_reset_sources() expects.

Signed-off-by: Paul Walmsley <paul@pwsan.com>
---
 arch/arm/mach-omap1/common.h | 2 +-
 arch/arm/mach-omap1/reset.c  | 7 ++++---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap1/common.h b/arch/arm/mach-omap1/common.h
index d6ac18d..ecd0bb6 100644
--- a/arch/arm/mach-omap1/common.h
+++ b/arch/arm/mach-omap1/common.h
@@ -94,6 +94,6 @@ extern int ocpi_enable(void);
 static inline int ocpi_enable(void) { return 0; }
 #endif
 
-extern int omap1_get_reset_sources(void);
+extern u32 omap1_get_reset_sources(void);
 
 #endif /* __ARCH_ARM_MACH_OMAP1_COMMON_H */
diff --git a/arch/arm/mach-omap1/reset.c b/arch/arm/mach-omap1/reset.c
index a0a9f97..cf8da1c 100644
--- a/arch/arm/mach-omap1/reset.c
+++ b/arch/arm/mach-omap1/reset.c
@@ -8,6 +8,7 @@
 
 #include <mach/hardware.h>
 
+#include "iomap.h"
 #include "common.h"
 
 /* ARM_SYSST bit shifts related to SoC reset sources */
@@ -43,12 +44,12 @@ void omap1_restart(char mode, const char *cmd)
  * Returns bits that represent the last reset source for the SoC.  The
  * format is standardized across OMAPs for use by the OMAP watchdog.
  */
-int omap1_get_reset_sources(void)
+u32 omap1_get_reset_sources(void)
 {
-	int ret = 0;
+	u32 ret = 0;
 	u16 rs;
 
-	rs = __raw_readw(ARM_SYSST);
+	rs = __raw_readw(OMAP1_IO_ADDRESS(ARM_SYSST));
 
 	if (rs & (1 << ARM_SYSST_POR_SHIFT))
 		ret |= 1 << OMAP_GLOBAL_COLD_RST_SRC_ID_SHIFT;
-- 
1.8.0

